### PR TITLE
Issue/3

### LINKED ISSUE #3: Update SDK for PHP 7.0 compat
Requires updating of the Campaign Monitor library. Lots of notices in the logs currently on PHP 7.4

### PR DESCRIPTION
**Pull request by mihaijoldis**
Updated SDK library and added the ConsentToTrack field which is set to Yes for new signups.
Closes #3

### CODE REVIEW

File: edd-campaign-monitor.php [hunk 1]
```diff
@@ -167,7 +168,7 @@ function eddcp_subscribe_fields() {
 
 // checks whether a user should be signed up for he mailchimp list
 function eddcp_check_for_email_signup($posted, $user_info) {
-	if($posted['eddcp_campaign_monitor_signup']) {
+	if( isset( $posted['eddcp_campaign_monitor_signup'] ) ) {
```
Comment (robincornett) [inline edd-campaign-monitor.php:171]: Thinking this should be an `empty` check instead of `isset`.

**Review by robincornett — COMMENTED** [review]
@ashleyfae wondering if this should be updated to use composer to pull in the SDK, and update to the current version (6.1.2; this is 6.1.0).

**Comment by ashleyfae** [comment]
@robincornett That would be lovely. With Mihai's permission you might want to handle that as I don't think he has Composer set up.

**Comment by ashleyfae** [comment]
This ready for review?

File: edd-campaign-monitor.php [hunk 2]
```diff
@@ -111,27 +114,22 @@ function eddcp_add_settings($settings) {
...
-		if( ! class_exists( 'CS_REST_Clients' ) )
-			require_once(EDDCP_PLUGIN_DIR . '/vendor/csrest_clients.php');
-
-		$wrap = new CS_REST_Clients($edd_options['eddcp_client'], $edd_options['eddcp_api']);
-		//echo '<pre>'; print_r( $wrap ); echo '</pre>'; exit;
-		$result = $wrap->get_lists();
+	$lists = array();
+	if ( empty( $edd_options['eddcp_api'] ) || empty( $edd_options['eddcp_client'] ) ) {
+		return $lists;
+	}
```
Comment (ashleyfae) [inline edd-campaign-monitor.php:122]: Early return. :heart:

**Review by ashleyfae — APPROVED** [review]
I reviewed it anyway. Pretty sure it was ready. Feel free to merge. Tested on https://github.com/awesomemotive/edd-campaign-monitor/pull/12, which uses this branch:

- [x] Lists fetched via API.
- [x] Signing up without opting in does not subscribe me.
- [x] Signing up with opting in does subscribe me.

**Pull request merged**